types/nsqjs/index.d.ts

```diff
@@ -48,11 +48,13 @@ export class Writer extends events.EventEmitter {
 
     constructor(nsqdHost: string, nsqdPort: number, options?: ConnectionConfigOptions);
 
-    connect(): any;
+    connect(): void;
 
-    publish(topic: string, msgs: any, listener?: (err: Error) => void): any;
+    publish(topic: string, msgs: string|Buffer|object|string[]|Buffer[]|object[], callback?: (err: Error) => void): void;
```

Comment:
I'm ok with this type for its documentation value, but it reduces to just `string | object` since `object` is a supertype of every constituent besides `string`.

Reply:
```suggestion
    publish(topic: string, msgs: string|Buffer|object|string[]|Buffer[]|object[], callback?: (err?: Error) => void): void;
```

I’m guessing `err` is not passed if publish is successful